platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/ExportMigrationEntry.java

```diff
@@ -0,0 +1,109 @@
+     * its value is blank
+     * @throws IllegalArgumentException if <code>name</code> is <code>null</code>
+     */
+    public default Optional<ExportMigrationEntry> getPropertyReferencedEntry(String name) {
+        return getPropertyReferencedEntry(name, (r, n) -> true);
+    }
+
+    /**
+     * Creates or retrieves (if already created) a migration entry referenced from the specified property
+     * in the properties file associated with this migration entry to be exported by the corresponding
+     * migratable.
+     * <p>
+     * The provided predicate will be invoked with the associated migration report and the property
+     * value (may be <code>null</code> if not defined). In case the property is not defined or is
+     * blank, an error will be automatically recorded unless the predicate returns <code>false</code>.
+     * Returning <code>true</code> in such case will still not create a corresponding migration entry.
+     * In all other cases, no errors or warning will be generated if the predicate returns <code>false</code>
+     * so it is up to the predicate to record one if required.
+     * <p>
+     * <i>Note:</i> The file referenced from the property is assumed to be relative to ${ddf.home} if
+     * not defined as absolute. All paths will automatically be relativized from ${ddf.home} if
+     * located underneath.
+     *
+     * @param name      the name of the property referencing a migration entry to create or retrieve
+     * @param validator a predicate to be invoked to validate the property value further which must
+     *                  return <code>true</code> to have a migration entry created
+     * @return a new migration entry or the existing one if already created for the referenced file
+     * from the specified property value or empty if the property is not defined, is blank or is invalid
+     * @throws IllegalArgumentException if <code>name</code> or <code>validator</code> is <code>null</code>
+     */
+    public Optional<ExportMigrationEntry> getPropertyReferencedEntry(String name,
+            BiPredicate<MigrationReport, String> validator);
+
+    /**
+     * Stores this entry's content in the export using the specified consumer based on this entry's
+     * path which can include sub-directories.
+     * <p>
+     * All errors and warnings are automatically recorded with the associated migration report including
+     * those thrown by the consumer logic.
+     *
+     * @param consumer a consumer capable of exporting the content of this entry to a provided output stream
+     * @return <code>true</code> if no errors were recorded as a result of processing this command;
+     * <code>false</code> otherwise
+     * @throws MigrationException       if a failure that prevents the operation from continue occurred
+     * @throws IllegalArgumentException if <code>consumer</code> is <code>null</code>
+     */
+    public boolean store(EBiConsumer<MigrationReport, OutputStream, IOException> consumer);
```

Comment:
I thought about it but after using it, this made it much simpler for the migratable. The reason is that when you use this approach, you will be working with the input/output stream directly and IOException are inevitable. Not having to worry about doing a try/catch for them makes it so much easier and so much simpler. I think with the EBiConsumer, we actually end up saving more code ;-) The EBiConsumer is a candidate for a utility library of our own or a good candidate for a library like Apache commons lang.